http_stubs/admin.py

```diff
@@ -22,21 +26,42 @@ def get_readonly_fields(self, *args, **kwargs) -> list:
         :param kwargs: optional kwargs
         :returns: list of fields.
         """
-        return [field.name for field in self.model._meta.fields]  # noqa:WPS437
+        fields = [field.name for field in self.model._meta.fields
+                  if field.name not in self.exclude]
+        return fields + list(self.readonly_fields)
 
     def has_add_permission(self, *args, **kwargs) -> bool:
-        """
-        Forbids adding new entries.
+        """Forbids adding new entries.
 
         :param args: optional args
         :param kwargs: optional kwargs
         :returns: False
         """
         return False
 
-    list_display = ('pk', 'date', 'http_stub', 'source_ip')
+    def raw_body(self, instance):
+        """Wrapped raw body."""
+        return mark_safe('<br>'.join(wrap(instance.body, width=240)))
+
+    raw_body.short_description = 'Raw request body'
+
+    def pretty_body(self, instance):
+        """Jsonify the request body if possible."""
+        try:
+            body = json.loads(instance.body)
+        except json.JSONDecodeError:
+            body = None
+        else:
+            body = json.dumps(body, indent=4)
+        body = body.replace(' ', '&nbsp;').replace('\n', '<br>')
```

Comment:
Maybe it would be easier to just wrap the text in `<pre>` tag?

Reply:
Yes, thanks